package eventql;

enum DHTNodeStatus {
  DHTNODE_JOINING = 0;
  DHTNODE_LIVE = 1;
  DHTNODE_MAINTENANCE = 2;
  DHTNODE_LEAVING = 3;
};

message DHTNode {
  required string name = 2;
  required string addr = 3;
  required DHTNodeStatus status = 4;
  repeated string sha1_tokens = 1;
};

message ClusterConfig {

  optional uint32 version = 1;

  repeated DHTNode dht_nodes = 3;
  optional uint32 dht_num_copies = 4 [default = 3];

};

enum ServerStatus {
  SERVER_DOWN = 0;
  SERVER_UP = 1;
};

message ServerConfig {
  optional uint64 version = 1;
  required string server_id = 2;
  optional string server_addr = 3;
  optional ServerStatus server_status = 4;
  repeated string sha1_tokens = 5;
  optional bool is_leaving = 6;
  optional bool is_dead = 7;
};

